File: dsaDotNet/Advanced/SubsetsAndSubSequences/Subset.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace dsaDotNet.Advanced.SubsetsAndSubSequences
{
     
    class Subset
    {
        public List<List<int>> subsets(List<int> A)
        {
            A.Sort();
            int n = (int)Math.Pow(2, A.Count);
            List<List<int>> ans = new List<List<int>>();
            for (int i = 0; i < n; i++)
            {
                List<int> li = new List<int>();
                for (int j = 0; j < A.Count; j++)
                {
                    if (((i) & (1 << j)) > 0) li.Add(A[j]);
                }
                ans.Add(li);
            }
            ans.Sort((x, y) => {
                int r = 0;
                for (int i = 0; i < Math.Min(x.Count, y.Count); i++)
                {
                    r=x[i].CompareTo(y[i]);
                    if(r!=0) return r;
                }
                r=x.Count.CompareTo(y.Count);
                return r;
                
            });
            return ans;
        }
    }

}

File: dsaJava/advanced/src/Subsets/subset.java
package Subsets;

import java.util.ArrayList;
import java.util.Collections;
import java.util.Comparator;

public class subset {
    public static void main(String[] args) {
        subset.subsets(new ArrayList<Integer>());
    }

        public static ArrayList<ArrayList<Integer>> subsets(ArrayList<Integer> A) {
            ArrayList<ArrayList<Integer>> answer=new ArrayList<>();
            Collections.sort(A);
            int index=-1;
            for(int i=0;i<Math.pow(2, A.size());i++){
                ArrayList<Integer> eles=new ArrayList<>();
                for(int j=0;j<A.size();j++){
                    if((i&(1<<j))!=0){
                        eles.add(A.get(j));
                    }
                }
                answer.add(eles);
            }
            Collections.sort(answer, new Comparator<ArrayList<Integer>>() {
                @Override
                public int compare(ArrayList<Integer> o1, ArrayList<Integer> o2) {
                    if(o1.size()>0 && o2.size()>0){
                        for(int i=0;i<Math.min(o1.size(), o2.size());i++){
                            if(o1.get(i)!=o2.get(i))
                                return o1.get(i)-o2.get(i);
                        }
                    }
                    return 0;
                }
            });
            return answer;
        }

}
